sklearn/dummy.py

```diff
@@ -433,6 +434,8 @@ def fit(self, X, y, sample_weight=None):
         self.n_outputs_ = y.shape[1]
 
         check_consistent_length(X, y, sample_weight)
+        if sample_weight is not None:
```

Comment:
I feel like this belongs to `check_array` too, doesn't it?

Reply:
not sure what you mean. Using ``check_array(sample_weights, ensure_2d=True)`` instead of asarray? We don't have any tests for NaN in sample weights, do we? I'm not sure how much I want to make this PR about adding way more checks to sample weights

Reply:
we have _check_sample_weight in validation

Reply:
that does a lot more though, right?